pkg/retry/retry.go

```diff
@@ -32,20 +32,43 @@ const (
 	DefaultMaxElapsedTime      = 1 * time.Minute
 )
 
-// Retry retries the given function DefaultRetryCount times while sleeping some time between unsuccessful attempts
-// if retryOnError returns true, continue retry, otherwise, return error
-func Do(operation func() error, retryOnError func(err error) bool) error {
-	defer trace.End(trace.Begin(""))
+// simplified config for configuring a back off object. Callers should populate and supply this to DoWithConfig
+type BackoffConfig struct {
+	InitialInterval     time.Duration
+	RandomizationFactor float64
+	Multiplier          float64
+	MaxInterval         time.Duration
 
-	var err error
-	var next time.Duration
+	// this field will indicate the maximum amount of "sleep" time that will occur.
+	MaxElapsedTime time.Duration
+}
 
```

Comment:
As we discussed over slack, I'm guessing you are going to introduce a new type called DefaultBackoffConfig with default values (or a NewBackoffConfig function that returns a config struct with defaults) so that caller can override the fields